File: go_app/entities/n1.go
package entities

import "time"

type N1 struct {
	CommonAccount
	Addr             string
	PK               string
	ReferralCode     string
	ReferralLocked   *time.Time
	FromReferralCode string
	ReferralTotal    int `gorm:"default:0"`
	Points           int `gorm:"default:0"`
	Rank             int `gorm:"default:0"`
	Faucet           *time.Time
	Usdc             int `gorm:"default:0"`
	DaliyAt          *time.Time
	UserID           int
	Amt              float64 `gorm:"default:0"`
	ReferralFlag     bool    `gorm:"default:0"`
	XToken           string  `gorm:"default:''"`
}

func (*N1) TableName() string {
	return "n1"
}

File: ts_app/schema/n1.ts
import {
  bigint,
  datetime,
  decimal,
  int,
  mysqlTable,
  tinyint,
  varchar,
} from "drizzle-orm/mysql-core";
import { baseColumns } from "./base_models";
import { InferSelectModel } from "drizzle-orm";

export const N1Table = mysqlTable(`n1`, {
  ...baseColumns(),
  addr: varchar("addr", { length: 256 }),
  pk: varchar("pk", { length: 256 }),
  amt: decimal("amt"),
  points: decimal("points"),
  rank: int("rank"),
  referralCode: varchar("referral_code", { length: 256 }),
  referral_locked: datetime("referral_locked"),
  fromReferralCode: varchar("from_referral_code", { length: 256 }),
  referralToal: int("referral_total"),
  daliyAt: datetime("daliy_at"),
  userID: int("user_id"),
  referralFlag: tinyint("referral_flag"),
  faucet: datetime("faucet"),
  xToken: varchar("x_token", { length: 256 }),
});

export type N1Account = InferSelectModel<typeof N1Table>;
